File: domain/src/main/kotlin/com/actiontracker/domain/model/ActionItem.kt
package com.actiontracker.domain.model

import kotlinx.serialization.Serializable

/**
 * A tracked, actionable task captured from shared content.
 *
 * IDs are client-generated UUIDs so items can be created offline without a
 * server round-trip. [preview] is present for [ContentType.LINK] items.
 *
 * A "wishlist" is not a special kind of item — it is simply a [Bucket] the user
 * created (e.g. named "Wishlist"). Items in such a bucket are ordinary action
 * items, so there is no dedicated wishlist flag or fields here.
 */
@Serializable
data class ActionItem(
    val id: String,
    val accountId: String,
    val bucketId: String,
    val title: String,
    val description: String? = null,
    val contentType: ContentType,
    val sourceContent: String? = null,
    val preview: LinkPreview? = null,
    val timeframe: Timeframe,
    val status: ActionStatus,
    val createdAt: Long,
    val sync: SyncMeta,
)

File: ios/SideQuestKit/Sources/SideQuestKit/Models/ActionItem.swift
import Foundation

/// A tracked, actionable task captured from shared content or manual entry
/// (contract: `backend/api/openapi.yaml` → `ActionItem`).
///
/// `id` is a client-generated UUID so items can be created offline without a
/// server round-trip (Req 5.7). `preview` is present for `ContentType.link`
/// items. `createdAt` orders items within a bucket on the board (Req 8.1) and
/// is a contract `date-time`, represented as a `Date` and encoded RFC 3339 (see
/// `SideQuestCoding`).
///
/// The on-the-wire field names are the contract's camelCase keys
/// (`accountId`, `bucketId`, `contentType`, `sourceContent`, `createdAt`), which
/// Swift's synthesized `Codable` produces directly, so no `CodingKeys` override
/// is required. Optional fields (`description`, `sourceContent`, `preview`) are
/// `nullable` in the contract.
public struct ActionItem: Codable, Identifiable, Equatable {

    public var id: String
    public var accountId: String
    public var bucketId: String
    public var title: String

    /// Optional, may be LLM-generated.
    public var description: String?

    public var contentType: ContentType

    /// Raw shared text / link / media reference.
    public var sourceContent: String?

    /// Present for link items; carries the resolved-or-raw preview (Req 4.9).
    public var preview: LinkPreview?

    public var timeframe: Timeframe
    public var status: ActionStatus

    /// Creation instant; orders items within a bucket (Req 8.1).
    public var createdAt: Date

    public var sync: SyncMeta

    public init(
        id: String,
        accountId: String,
        bucketId: String,
        title: String,
        description: String? = nil,
        contentType: ContentType,
        sourceContent: String? = nil,
        preview: LinkPreview? = nil,
        timeframe: Timeframe,
        status: ActionStatus,
        createdAt: Date,
        sync: SyncMeta
    ) {
        self.id = id
        self.accountId = accountId
        self.bucketId = bucketId
        self.title = title
        self.description = description
        self.contentType = contentType
        self.sourceContent = sourceContent
        self.preview = preview
        self.timeframe = timeframe
        self.status = status
        self.createdAt = createdAt
        self.sync = sync
    }
}
